src/Server/TCPHandler.cpp

```diff
@@ -902,12 +901,13 @@ AsynchronousInsertQueue::PushResult TCPHandler::processAsyncInsertQuery(Asynchro
         }
     }
 
-    auto planned_chunk = squashing.flush();
-    Chunk result_chunk;
-    if (planned_chunk.hasChunkInfo())
-        result_chunk = DB::Squashing::squash(std::move(planned_chunk));
+    Chunk result_chunk = Squashing::squash(squashing.flush());
```

Comment:
@yariks5s I think you said that this `squash` is not necessary here, but I think it is. It might happen that `result_chunk` has a `ChunksToMerge` chunk info, so we need to squash them to extract the data from those chunks.

Reply:
Yes, I thought that `flush` was already doing this job, I confused the versions